src/RestWrite.js

```diff
@@ -1557,10 +1557,22 @@ RestWrite.prototype.runDatabaseOperation = function () {
 
         // Quick check, if we were able to infer the duplicated field name
         if (error && error.userInfo && error.userInfo.duplicated_field === 'username') {
-          throw new Parse.Error(
+          //Instead of directly throwing the error, we will give error based on logEvent
+          const usernameError = new Parse.Error(
             Parse.Error.USERNAME_TAKEN,
             'Account already exists for this username.'
           );
+
+          // Check for custom log level
+          const logLevel = this.config.logEvents && this.config.logEvents.usernameAlreadyExists;
+          if (logLevel) {
+            // Use custom log level
+            this.config.loggerController[logLevel](JSON.stringify(usernameError));
+          } else if (logLevel !== false) {
+            // Use default error logging if not explicitly disabled
+            this.config.loggerController.error(JSON.stringify(usernameError));
+          }
+          throw usernameError;
```

Comment:
This logic should not be necessary, there is always a default value that is defined in the Options definition.

This should be enough:
`this.config.loggerController[ this.config.logLevels.usernameAlreadyExists](JSON.stringify(usernameError));`